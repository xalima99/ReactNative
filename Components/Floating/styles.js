import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    height: 65,
    padding: 16,
    width: 65,
    position: "absolute",
    borderRadius: 50,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#020191",
    color: "#fff",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,

    elevation: 5,
  },
  secondary: {
    backgroundColor: "#fff",
  },
  primaryDisabled: {
    backgroundColor: "#E5E5E5",
  },
  iconColor: {
    tintColor: "#fff",
  },
  cont: {
    height: "100%",
    width: "100%",
  },
});
