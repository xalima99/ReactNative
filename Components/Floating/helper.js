import styles from "./styles";

export const getTheme = (primary, secondary, disabled) => {
  const containerTheme = [styles.container];

  if (secondary) {
    containerTheme.push(styles.secondary);
  }
  if (disabled) {
    containerTheme.push(styles.primaryDisabled);
  }
  return containerTheme;
};
