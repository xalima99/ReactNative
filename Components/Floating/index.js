import React, { useMemo } from "react";
import { Image, TouchableOpacity } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import PropTypes from "prop-types";
import { getTheme } from "./helper";

const index = ({
  right = 20,
  bottom = 20,
  onPress,
  icon = "plus",
  image,
  secondary,
  disabled,
}) => {
  const ContainerTheme = useMemo(() => getTheme(true, secondary, disabled), [
    secondary,
    disabled,
  ]);

  return (
    <TouchableOpacity
      activeOpacity={0.9}
      disabled={disabled}
      onPress={onPress}
      style={[
        {
          backgroundColor: "#010273",
        },
        ...ContainerTheme,
        { right, bottom },
      ]}
    >
      {image ? (
        <Image
          source={image}
          style={{ tintColor: "#fff", height: 30, width: 30 }}
        />
      ) : (
        <MaterialCommunityIcons
          name={icon}
          color={secondary ? "#020191" : disabled ? "#CFCFCF" : "#fff"}
          size={35}
        />
      )}
    </TouchableOpacity>
  );
};

index.propTypes = {
  right: PropTypes.number,
  bottom: PropTypes.number,
  onPress: PropTypes.func,
  icon: PropTypes.string,
  image: PropTypes.any,
  secondary: PropTypes.bool,
  disabled: PropTypes.bool,
};

export default index;
