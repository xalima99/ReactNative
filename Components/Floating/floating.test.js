import React from "react";
import { render } from "@testing-library/react-native";

import Floating from "./index";

it("Tests Snapshot", () => {
  const snap = render(<Floating />).toJSON();
  expect(snap).toMatchSnapshot();
});
