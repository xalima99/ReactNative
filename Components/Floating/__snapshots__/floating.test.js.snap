// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#020191",
      "borderRadius": 50,
      "bottom": 20,
      "color": "#fff",
      "elevation": 5,
      "height": 65,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 16,
      "position": "absolute",
      "right": 20,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
      "width": 65,
    }
  }
>
  <Text />
</View>
`;
